GitUI/UserControls/RevisionGrid/Columns/RevisionGraphColumnProvider.cs

```diff
@@ -551,4 +510,94 @@ string GetLaneInfo(int x, int rowIndex)
             }
         }
     }
+
+    public sealed class GraphCache
+    {
+        [CanBeNull] private Bitmap _graphBitmap;
+        [CanBeNull] private Graphics _graphBitmapGraphics;
+        private int _cacheCount;
+        private int _cacheCountMax;
+        private int _cacheHead = -1;
+        private int _cacheHeadRow;
+
+        public Bitmap GraphBitmap => _graphBitmap;
+        public Graphics GraphBitmapGraphics => _graphBitmapGraphics;
+
+        /// <summary>
+        /// The 'slot' that is the head of the circular bitmap.
+        /// </summary>
+        public int Head
+        {
+            get => _cacheHead;
+            set => _cacheHead = value;
+        }
+
+        /// <summary>
+        /// The node row that is in the head slot.
+        /// </summary>
+        public int HeadRow
+        {
+            get => _cacheHeadRow;
+            set => _cacheHeadRow = value;
+        }
+
+        /// <summary>
+        /// Number of elements in the cache.
+        /// </summary>
+        public int Count
+        {
+            get => _cacheCount;
+            set => _cacheCount = value;
+        }
+
+        /// <summary>
+        /// Number of elements allowed in the cache. Is based on control height.
+        /// </summary>
+        public int CountMax
+        {
+            get => _cacheCountMax;
+            set => _cacheCountMax = value;
+        }
+
+        public void Resize(int width, int height, int laneWidth)
+        {
+            if (_graphBitmap != null && _graphBitmap.Width >= width && _graphBitmap.Height == height)
+            {
+                return;
+            }
+
+            if (_graphBitmap != null)
+            {
+                _graphBitmap.Dispose();
+                _graphBitmap = null;
+            }
+
+            if (_graphBitmapGraphics != null)
+            {
+                _graphBitmapGraphics.Dispose();
+                _graphBitmapGraphics = null;
+            }
+
+            _graphBitmap = new Bitmap(
+                Math.Max(width, laneWidth * 3),
+                height,
+                PixelFormat.Format32bppPArgb);
+            _graphBitmapGraphics = Graphics.FromImage(_graphBitmap);
+            _graphBitmapGraphics.SmoothingMode = SmoothingMode.AntiAlias;
+            _cacheHead = 0;
+            _cacheCount = 0;
+        }
+
+        public void Clear()
+        {
+            _cacheHead = -1;
+            _cacheHeadRow = 0;
+        }
+
+        public void Reset()
+        {
+            _cacheHead = 0;
+            _cacheCount = 0;
+        }
```

Comment:
after the refactor it now very obvious that `Clear` and `Reset` are very similar, yet I am unable to fully grasp why we need both.
perhaps you can help